zfs-replicate.sh

```diff
@@ -1,524 +1,538 @@
 #!/usr/bin/env bash
 ## zfs-replicate.sh
+set -eo pipefail ## fail fast
 
 ############################################
 ##### warning gremlins live below here #####
 ############################################
 
+## define read-only constants
+readonly MODE_PUSH="PUSH"
+readonly MODE_PULL="PULL"
+readonly TYPE_REMOTE="REMOTE"
+readonly TYPE_LOCAL="LOCAL"
+
```

Comment:
I'm not sure why the Github actions wants to change the REMOTE and LOCAL to TYPE_REMOTE and TYPE_LOCAL. It makes no sense. The variable is set via the `config.sh` file.

Reply:
Also not really sure why it has to set the variables to readonly.